.pre-commit-config.yaml

```diff
@@ -15,6 +15,7 @@ repos:
         args:
           - --extend-ignore=E501
           - --extend-select=B950
+          - --per-file-ignores=./bagelbids/cli.py:F841
```

Comment:
Sounds good (and apologies if we have discussed this already) but could you not just add this directly to the project config (https://github.com/neurobagel/bagelbids/blob/main/setup.cfg#L57).

Reply:
So this is actually kinda tricky to answer and I'm still not 100% sure but from what I can tell, if I changed this Flake8 option in setup.cfg alone, pre-commit doesn't know about it automatically. The current issue is that pre-commit itself prevents you from committing if you have local variables that are assigned but not used (due to rule F841) when you run `git commit` - so it's not really working exactly the same as a normal linter. I know pre-commit has a command-line option `--config` where you can specify a custom config file, but running `pre-commit install --config setup.cfg` (with the --per-file-ignores rule in setup.cfg) or doing something similar to this [answer](https://github.com/pre-commit/pre-commit/issues/1723#issuecomment-738777992) directly in the pre-commit config file doesn't seem to work right. My hunch is that if I rely purely on setup.cfg to define the pre-commit config (if this is even possible - the dev is famously opposed to supporting pre-commit config entirely in Python project configuration files lol, see [here](https://github.com/pre-commit/pre-commit/issues/1733) and [here](https://github.com/pre-commit/pre-commit/issues/1165)) I would probably have to re-run `pip install -e` which would walk through the entire project setup again whenever I wanted to change something with pre-commit. Meanwhile, it's very easy to just make a change in pre-commit-config.yaml and apply the changes with `pre-commit install` - you don't have to touch anything else in the python project.

Not sure if that made a whole lot of sense but basically what I've gleaned is that the `pre-commit` dev optimized the package to work as independently as possible in a project - so I'm happy to rely on pre-commit-specific files to do this kind of thing

Reply:
oh my, thanks for the deep dive. It seems that if the dev doesn't want to support it, we should probably not rely on a workaround anyway. So let's just keep the flake8 rules in two places and keep an eye on keeping them in sync. There aren't that many rules anyway.